Syntax error in mock function when allOf and nested schema references are used together

<!-- Thanks for reporting an issue! Please fill out the blanks below. -->

## What are the steps to reproduce this issue?

1. Generated with below

### orval.config.ts

```ts
import { defineConfig } from "orval";
export default defineConfig({
  test: {
    input: {
      target: "./test.yaml",
    },
    output: {
      workspace: "./generated",
      target: "functions",
      client: "react-query",
      mock: true,
    },
  },
});
```

### test.yaml

```yaml
openapi: 3.0.3
info:
  version: 1.0.0
  title: Test
paths:
  /users:
    get:
      operationId: getUsers
      responses:
        "200":
          description: OK
          content:
            application/json:
              schema:
                allOf:
                  - $ref: "#/components/schemas/User"
                  - $ref: "#/components/schemas/Meta"
components:
  schemas:
    User:
      type: object
      properties:
        name:
          type: string
    Users:
      type: array
      items:
        $ref: "#/components/schemas/User"
    Meta:
      $ref: "#/components/schemas/MetaData"
    MetaData:
      type: object
      properties:
        total:
          type: integer
```

### command

```bash
> pnpm orval
🍻 Start orval v6.23.0 - A swagger client generator for typescript
⚠️  test - Prettier not found
🎉 test - Your OpenAPI spec has been converted into ready to use orval!
```

## What happens?

generated functions with below mock function with syntax errors

### generated/functions/test.ts

```ts
export const getGetUsersMock = () => ({name: faker.helpers.arrayElement([faker.word.sample(), undefined]),{total: faker.helpers.arrayElement([faker.number.int({min: undefined, max: undefined}), undefined])}})
```

### tsc errors

```
> pnpm tsc         
generated/functions/test.ts:103:107 - error TS1136: Property assignment expected.

103 export const getGetUsersMock = () => ({name: faker.helpers.arrayElement([faker.word.sample(), undefined]),{total: faker.helpers.arrayElement([faker.number.int({min: undefined, max: undefined}), undefined])}})
                                                                                                              ~

generated/functions/test.ts:103:120 - error TS1005: ',' expected.

103 export const getGetUsersMock = () => ({name: faker.helpers.arrayElement([faker.word.sample(), undefined]),{total: faker.helpers.arrayElement([faker.number.int({min: undefined, max: undefined}), undefined])}})
                                                                                                                           ~

generated/functions/test.ts:103:128 - error TS1005: ',' expected.

103 export const getGetUsersMock = () => ({name: faker.helpers.arrayElement([faker.word.sample(), undefined]),{total: faker.helpers.arrayElement([faker.number.int({min: undefined, max: undefined}), undefined])}})
                                                                                                                                   ~

generated/functions/test.ts:103:141 - error TS1005: ',' expected.

103 export const getGetUsersMock = () => ({name: faker.helpers.arrayElement([faker.word.sample(), undefined]),{total: faker.helpers.arrayElement([faker.number.int({min: undefined, max: undefined}), undefined])}})
                                                                                                                                                ~

generated/functions/test.ts:103:206 - error TS1128: Declaration or statement expected.

103 export const getGetUsersMock = () => ({name: faker.helpers.arrayElement([faker.word.sample(), undefined]),{total: faker.helpers.arrayElement([faker.number.int({min: undefined, max: undefined}), undefined])}})
                                                             
...
```

## What were you expecting to happen?

The following functions were obtained by eliminating the "Meta" nesting.
The syntax seems to be correct.

```diff
-                  - $ref: "#/components/schemas/Meta"
+                  - $ref: "#/components/schemas/MetaData"
```

```bash
❯ pnpm orval
🍻 Start orval v6.23.0 - A swagger client generator for typescript
🎉 test - Your OpenAPI spec has been converted into ready to use orval!
```

```ts
export const getGetUsersMock = () => ({
  name: faker.helpers.arrayElement([faker.word.sample(), undefined]),
  total: faker.helpers.arrayElement([
    faker.number.int({ min: undefined, max: undefined }),
    undefined,
  ]),
});
```

## Any logs, error output, etc?

…

## Any other comments?

…

## What versions are you using?

**Operating System:**
Ubuntu 22.04
```bash
> uname -a                                                  
Linux deve 5.15.0-89-generic #99-Ubuntu SMP Mon Oct 30 20:42:41 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
> cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
```

**Package Version:**
orval 6.23.0
```bash
> cat package.json| grep orval
    "orval": "^6.23.0",
```
**Browser Version:**
None

--- a/packages/mock/src/faker/getters/combine.ts
+++ b/packages/mock/src/faker/getters/combine.ts
@@ -116,6 +116,14 @@ export const combineSchemasMock = ({
       currentValue = `${currentValue ? `${currentValue},` : ''}${itemResolvedValue.value}`;
     }
 
+    if (
+      resolvedValue.type === undefined &&
+      currentValue &&
+      separator === 'allOf'
+    ) {
+      currentValue = `...${currentValue}`;
+    }
+
     const isObjectBounds =
       !combine ||
       (['oneOf', 'anyOf'].includes(combine.separator) && separator === 'allOf');

--- a/tests/specifications/all-of.yaml
+++ b/tests/specifications/all-of.yaml
@@ -27,6 +27,22 @@ paths:
                   - type: object
                     required:
                       - color
+  /rested-ref-in-all-of-pets:
+    get:
+      operationId: getNestedRefInAllOfPets
+      tags:
+        - pets
+      description: |-
+        Nested ref in allOf pets.
+      responses:
+        '200':
+          description: User
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: '#/components/schemas/PetRef'
+                  - $ref: '#/components/schemas/PetDetail'
 
 components:
   schemas:
@@ -48,3 +64,5 @@ components:
       properties:
         tag:
           type: string
+    PetRef:
+      $ref: '#/components/schemas/Pet'

